packages/flutter/lib/src/gestures/converter.dart

```diff
@@ -104,90 +93,44 @@ class PointerEventConverter {
           );
           break;
         case ui.PointerChange.hover:
-          final bool alreadyAdded = _pointers.containsKey(datum.device);
+        case ui.PointerChange.move:
```

Comment:
I'd keep the cases for hover and move completely separate since they don't share too much code.